shared-bin/src/logger.rs

```diff
@@ -126,26 +116,26 @@ pub fn init(
         None
     };
 
-    let mut error_only = vec![
-        span_names::OLD_BROADCAST_OFFERS,
-        xtras::send_interval::QUIET_NAME,
-    ];
-
-    if !verbose_spans {
-        error_only.extend([
-            xtra_libp2p_ping::ping::PING_PEER_SPAN,
-            xtra_libp2p_offer::maker::LIBP2P_BROADCAST_OFFERS_SPAN,
-        ]);
-    }
-
-    let disable_spans = tracing_subscriber::filter::filter_fn(move |meta| {
-        !(*meta.level() > Level::ERROR && error_only.contains(&meta.name()))
+    let disable_spans = dynamic_filter_fn(move |meta, ctx| {
+        if meta.is_span() && !["tokio", "runtime"].contains(&meta.target()) {
+            match ctx.lookup_current() {
+                Some(parent) => {
+                    !(parent.name() == "always_quiet_children"
```

Comment:
I suspect `always_quiet_children` could be a const in this module (important for maintainability). same goes for the other one.

Reply:
fair enough! I didn't notice that xtra-libp2p-ping needs to depend on it. in such case, I would create a small separate crate out of it indeed.